### PR TITLE
Use block-based animations

### PR DESCRIPTION
**Pull request by awe-mt**
The other APIs are deprecated in iOS 13.

### CODE REVIEW

File: MBProgressHUD.m [hunk 1]
```diff
@@ -324,13 +324,12 @@ - (void)showUsingAnimation:(BOOL)animated {
...
-		[UIView setAnimationDuration:0.30];
-		self.alpha = 1.0f;
-		if (animationType == MBProgressHUDAnimationZoomIn || animationType == MBProgressHUDAnimationZoomOut) {
-			self.transform = rotationTransform;
-		}
-		[UIView commitAnimations];
+    [UIView animateWithDuration:0.30 animations:^{
+      self.alpha = 1.0f;
+      if (self.animationType == MBProgressHUDAnimationZoomIn || self.animationType == MBProgressHUDAnimationZoomOut) {
+        self.transform = self->rotationTransform;
```
Comment (awe-mt) [inline MBProgressHUD.m:330]: > self->rotationTransform

Yep, it's a struct... 😭

File: MBProgressHUD.m [hunk 2]
```diff
@@ -735,30 +731,19 @@ - (void)updateUIForKeypath:(NSString *)keyPath {
...
 }
 
-#pragma mark - Notifications
+#pragma mark - Handle orientation changes
 
-- (void)registerForNotifications {
-	NSNotificationCenter *nc = [NSNotificationCenter defaultCenter];
-
-	[nc addObserver:self selector:@selector(statusBarOrientationDidChange:)
-			   name:UIApplicationDidChangeStatusBarOrientationNotification object:nil];
```
Comment (mt-erik) [inline MBProgressHUD.m:744]: Shouldn't we just [use](https://developer.apple.com/documentation/uikit/uideviceorientationdidchangenotification) the `UIDeviceOrientationDidChangeNotification` instead?

Comment (awe-mt) [reply MBProgressHUD.m:744]: Yes, if that's available that sounds like a much less risky chance to me. Let's do that!

**Comment by mt-erik** [comment]
This was replaced by #11

**Pull request closed without merging**